### DISCUSSION TITLE
Graphics OLED library possible?

### DISCUSSION THREAD

**Discussion by mechatroniks-git** [SpenceKonde/megaTinyCore#454] [Q&A] [answered]
I've tried TINY4KOLED, only 2 fonts, none that I liked. U8x8 seems to work well with more fonts

I was hoping I could get U8G2. It appears not to compile and suggests U8x8. 

Is that expected or has anyone successfully implemented U8G2 on Series 0/1/2 ATtiny chips?

**Comment by grandaspanna** [comment, marked as answer]
I've successfully used u8g2 on a 1614 project with a cheap 128x64 OLED. Graphics, bitmaps, fonts all seem to work ok.

Fonts are memory-intensive, so choose carefully.

**Comment by SpenceKonde** [comment]
When a library doesn't compile on this core can you please post the full text of any and all compile errors, as well as the part you were trying to compile for? I can't provide any insight without more information.

**Comment by SpenceKonde** [comment]
And you say U8x8 works here, can you let me know a URL or reference for that so I can add it to the Library Compatibility page?

Same goes for U82G, if we can figure out why it works for some but not others?

**Reply by grandaspanna** [reply]
Don't know about OP, but I'm using this: https://github.com/olikraus/u8g2, installed via library manager. I'm on 2.27.6 (one release behind current from looks of things)

**Comment by mechatroniks-git** [comment]
Hey @SpenceKonde, the U8x8 that works is here: https://github.com/olikraus/u8g2  v2.28.10

Now that I know the U8G2 is possible, I'll go back to it and try a bit harder. I'll post the code & errors also.

**Comment by mechatroniks-git** [comment]
Thanks @grandaspanna, I was able to get U8G2 working. I'm running at 3.3V, so 10MHz. Once I got past the compile errors, it worked. 

Sample code for U8x8 [here](https://gist.github.com/mechatroniks-git/0b3486c3bad5af1bc2ee71d52c34c7e5)

Sample code for U8G2 [here](https://gist.github.com/mechatroniks-git/b5ec451ec4b610318717184ac980b37a)

**Comment by SpenceKonde** [comment]
Thanks much for info I have updated tcompatible libraries chart!
